### ISSUE TITLE
False positive for conditional_returns_on_newline with returnKeyType

### ISSUE THREAD

**Issue by mattrubin** [realm/SwiftLint#784]
The `conditional_returns_on_newline` rule returns a false positive for the following line:
```swift
if textFieldCell.textField.returnKeyType == .Next {
```
It looks like [the regex used by the rule](https://github.com/realm/SwiftLint/blob/050ccd17972e247571d2a494708eb0a7df59acb7/Source/SwiftLintFramework/Rules/ConditionalReturnsOnNewline.swift#L36) simply checks for the word "return" appearing on the same line as an `if` or `guard` and doesn't ensure that the "return" string is the `return` keyword.

I'm not certain how to improve the regex. Checking for a whitespace character after "return" is not sufficient since `return(true)` and `return[]` are – though ugly – technically valid Swift.

**Comment by jpsim** [comment]
Ah, I should've caught this in code review. `return` should be checked to be a built-in syntax type, rather than an identifier, as it is in your example. Thanks for reporting this @mattrubin!

**Comment by nikolay-dementiev** [comment]
regex line should be something like this: (guard|if)[^\n]*([\s]return)[^\n]\n* (I added "[\s]" before return keyword) 

jpsim I'm Right?
